FinSearchUnified/Tests/Helper/UrlBuilderTest.php

```diff
@@ -83,4 +114,86 @@ private function setClientIp($ipAddress)
     {
         $_SERVER['HTTP_CLIENT_IP'] = $ipAddress;
     }
+
+    /**
+     * @return array
+     */
+    public function forceOriginalQueryProvider()
+    {
+        return [
+            'forceOriginalQuery not present' => [null],
+            'forceOriginalQuery present and truthy' => [1],
+            'forceOriginalQuery present and falsy' => [0],
+        ];
+    }
+
+    /**
+     * @dataProvider forceOriginalQueryProvider
+     *
+     * @param int|null $forceOriginalQuery
+     *
+     * @throws \Exception
+     */
+    public function testBuildUrlAndResponse($forceOriginalQuery)
+    {
+        $sessionArray = [
+            ['isSearchPage', true]
+        ];
+        // Create mock object for Shopware Session and explicitly return the values
+        $session = $this->getMockBuilder('\Enlight_Components_Session_Namespace')
+            ->setMethods(['offsetGet'])
+            ->getMock();
+        $session->expects($this->atLeastOnce())
+            ->method('offsetGet')
+            ->willReturnMap($sessionArray);
+
+        // Assign mocked session variable to application container
+        Shopware()->Container()->set('session', $session);
+
+        $_GET['forceOriginalQuery'] = $forceOriginalQuery;
+
+        $urlBuilder = new UrlBuilder($this->httpClient);
+
+        // Create criteria object with necessary conditions
+        $criteria = new Criteria();
+        $criteria->offset(0)->limit(2)
+            ->addBaseCondition(new SearchTermCondition('findologic'))
+            ->addBaseCondition(new CategoryCondition([Shopware()->Shop()->getCategory()->getId()]))
+            ->addSorting(new ProductNameSorting());
+
+        $urlBuilder->buildQueryUrlAndGetResponse($criteria);
+
+        $requestedUrl = $this->httpClient->getUri()->getQueryAsArray();
+        $url = $this->httpClient->getUri();
+        $actualUrl = sprintf('%s://%s%s', $url->getScheme(), $url->getHost(), $url->getPath());
+
+        $this->assertSame(
+            sprintf('https://service.findologic.com/ps/xml_2.0/%s/index.php', Shopware()->Shop()->getHost()),
+            $actualUrl,
+            'The resulting URL is not correct'
+        );
+        $this->assertArrayHasKey(
+            'userip',
+            $requestedUrl,
+            'userip was not found in the query parameters'
```

Comment:
Wrap `userip`, `revision` etc. with double quotes, so it is clear that this is the actualy name of the parameter.

Reply:
The first parameter of `assertArrayHasKey` can be in single quotes. What I meant was that the assertion message should look something like `"userip" was not found in the query parameters`.